### PR TITLE
fix: unauthorized error in organization

### LINKED ISSUE #11038: Unauthorized error in organization creation step
Screenshot 2023-08-31 at 2 38 44 PM

1) Go to http://localhost:3000/settings/organizations/new
2) Create organization and set password in step 1 and step 2

### PR DESCRIPTION
**Pull request by Udit-takkar**
## What does this PR do?

Fixes https://github.com/calcom/cal.com/issues/11038

https://github.com/calcom/cal.com/assets/53316345/f00c31e2-4938-4b38-b433-66a7b5a900df

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: packages/trpc/server/routers/viewer/organizations/update.handler.ts [hunk 1]
```diff
@@ -24,10 +23,7 @@ export const updateHandler = async ({ ctx, input }: UpdateOptions) => {
   // A user can only have one org so we pass in their currentOrgId here
   const currentOrgId = ctx.user?.organization?.id || input.orgId;
 
-  if (!currentOrgId || ctx.user.role !== UserPermissionRole.ADMIN)
-    throw new TRPCError({ code: "UNAUTHORIZED" });
-
-  if (!(await isOrganisationAdmin(ctx.user?.id, currentOrgId)) || ctx.user.role !== UserPermissionRole.ADMIN)
+  if (!currentOrgId || !(await isOrganisationAdmin(ctx.user?.id, currentOrgId)))
```
Comment (Udit-takkar) [inline packages/trpc/server/routers/viewer/organizations/update.handler.ts:26]: organization admin should be able to update organization.  ctx.user.role is different from organization's role

Comment (leog) [reply packages/trpc/server/routers/viewer/organizations/update.handler.ts:26]: I think that was there to let the instance admin be able to do whatever in an organization too

**Review by leog — CHANGES_REQUESTED** [review]
Left a comment to agree on behavior before merging this or not

File: packages/trpc/server/routers/viewer/organizations/update.handler.ts (hunk 1 above)
Comment (Udit-takkar) [reply packages/trpc/server/routers/viewer/organizations/update.handler.ts]: okay. do we have to check if the instance admin is atleast member of organization ? Like

`
const isUserRoleAdmin = ctx.user.role === UserPermissionRole.ADMIN && (await isOrganisationMember(ctx.user?.id, currentOrgId))
`

File: packages/trpc/server/routers/viewer/organizations/update.handler.ts [hunk 2]
```diff
@@ -24,11 +24,12 @@ export const updateHandler = async ({ ctx, input }: UpdateOptions) => {
   // A user can only have one org so we pass in their currentOrgId here
   const currentOrgId = ctx.user?.organization?.id || input.orgId;
 
-  if (!currentOrgId || ctx.user.role !== UserPermissionRole.ADMIN)
-    throw new TRPCError({ code: "UNAUTHORIZED" });
+  const isUserOrganizationAdmin = currentOrgId && (await isOrganisationAdmin(ctx.user?.id, currentOrgId));
+  const isUserRoleAdmin = ctx.user.role === UserPermissionRole.ADMIN;
```
Comment (Udit-takkar) [inline packages/trpc/server/routers/viewer/organizations/update.handler.ts:28]: https://github.com/calcom/cal.com/pull/11039#discussion_r1311709557

**Review by leog — APPROVED** [review]
LGTM thanks for taking care of this

**Pull request merged**